applet/src/main/java/jpake/jpakePassivePayload.java

```diff
@@ -0,0 +1,213 @@
+        output[currStart] = (byte)senderID.length;
+        currStart +=1;
+        Util.arrayCopyNonAtomic(senderID, (short)0, output, currStart, (short)senderID.length);
+        currStart += senderID.length;
+        output[currStart] = (byte)encodedG1.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(encodedG1, (short) 0, output, currStart, (short)encodedG1.length);
+        currStart += encodedG1.length;
+        output[currStart] = (byte)encodedG2.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(encodedG2, (short) 0, output, currStart, (short)encodedG2.length);
+        currStart += encodedG2.length;
+        output[currStart] = (byte)encA.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(encA, (short) 0, output, currStart, (short)encA.length);
+        currStart += encA.length;
+        output[currStart] = (byte)publicA1Encoded.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(publicA1Encoded, (short) 0, output, currStart, (short)publicA1Encoded.length);
+        currStart += publicA1Encoded.length;
+        output[currStart] = (byte)publicV1Encoded.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(publicV1Encoded, (short) 0, output, currStart, (short)publicV1Encoded.length);
+        currStart += publicV1Encoded.length;
+        output[currStart] = (byte)result1.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(result1, (short) 0, output, currStart, (short)result1.length);
+        currStart += result1.length;
+        output[currStart] = (byte)publicA2Encoded.length;
+        currStart +=1;
+        Util.arrayCopyNonAtomic(publicA2Encoded, (short) 0, output, currStart, (short)publicA2Encoded.length);
+        currStart += publicA2Encoded.length;
+        output[currStart] = (byte) publicV2Encoded.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(publicV2Encoded, (short) 0, output, currStart, (short)publicV2Encoded.length);
+        currStart += publicV2Encoded.length;
+        output[currStart] = (byte)result2.length;
+        currStart +=1;
+        Util.arrayCopyNonAtomic(result2, (short) 0, output, currStart, (short)result2.length);
+        currStart += result2.length;
+        output[currStart] = (byte)publicA3Encoded.length;
+        currStart +=1;
+        Util.arrayCopyNonAtomic(publicA3Encoded, (short) 0, output, currStart, (short)publicA3Encoded.length);
+        currStart += publicA3Encoded.length;
+        output[currStart] = (byte) publicV3Encoded.length;
+        currStart += 1;
+        Util.arrayCopyNonAtomic(publicV3Encoded, (short) 0, output, currStart,(short)publicV3Encoded.length);
+        currStart += publicV3Encoded.length;
+        output[currStart] = (byte)result3.length;
+        currStart +=1;
+        Util.arrayCopyNonAtomic(result3, (short) 0, output, currStart,(short)result3.length);
+        return output;
+    }
+    public static jpakePassivePayload fromBytes(byte[] input){
```

Comment:
please add `throws` clause for cases when `input` is not a valid representation of `jpakePassivePayload `

Reply:
Done.